casper/src/main/scala/coop/rchain/casper/api/BlockApiImpl.scala

```diff
@@ -174,11 +179,63 @@ class BlockApiImpl[F[_]: Concurrent: RuntimeManager: BlockDagStorage: BlockStore
       )
   }
 
-  override def deployStatus(deployId: DeployId): F[ApiErr[String]] =
-    BlockDagStorage[F]
-      .asInstanceOf[BlockDagKeyValueStorage[F]]
-      .deployStatus(deployId)
-      .map(_.asRight[Error])
+  override def deployStatus(deployId: DeployId): F[ApiErr[DeployExecStatus]] = {
+
+    def deployBlockHash: F[String] =
+      findDeploy(deployId).map(_.getOrElse(LightBlockInfo.defaultInstance)).map(_.blockHash)
+
+    def getBlockInfo(blockHash: String): F[BlockInfo] =
+      getBlock(blockHash).map(_.getOrElse(BlockInfo.defaultInstance))
+
+    def getSeqPar(blockHash: String): F[Seq[Par]] =
+      for {
+        par       <- Sync[F].delay(GDeployId(deployId).asInstanceOf[Par])
+        dataAtPar <- getDataAtPar(par, blockHash, usePreStateHash = false)
+        seqPar    <- dataAtPar.leftMap(new Exception(_)).liftTo[F].map { case (seqPar, _) => seqPar }
+      } yield seqPar
+
+    def withBlockInfoAndSeqPar[A](f: (BlockInfo, Seq[Par]) => A) =
+      for {
+        blockHash <- deployBlockHash
+        blockInfo <- getBlockInfo(blockHash)
+        seqPar    <- getSeqPar(blockHash)
+      } yield f(blockInfo, seqPar)
+
+    def notProcessed(message: String) = DeployExecStatus().withNotProcessed(NotProcessed(message))
+
+    for {
+      status      <- BlockDagStorage[F].asInstanceOf[BlockDagKeyValueStorage[F]].deployStatus(deployId)
```

Comment:
This is not safe cast. Better to accept `BlockDagKeyValueStorage` or just deploy status function.

Reply:
@tgrospic, the simplest solution is to move `deployStatus` to `BlockDagStorage` trait as I mentioned [before](https://github.com/rchain/rchain/pull/3682#discussion_r871296148).

Reply:
Repeating my comment form PR you mentioned.

`deployStatus` which returns string is just temporary functions and should not be on block dag storage at all.

We need to see what is needed for BlockApi to return deploy status and add just that to block dag storage.